`ethereum.base_types.MAX_VALUE` isn't a real constant

gurukamath opened the pull request:
### What was wrong?
The rendered documentation is showing [MAX_VALUE](https://ethereum.github.io/execution-specs/autoapi/ethereum/base_types/index.html#ethereum.base_types.MAX_VALUE), but that constant isn't defined in the source code.

Related to Issue #319 

### How was it fixed?
A couple of things were fixed.

- `autoapi` seems to not be able to capture the value assignment of a class attribute. This has been patched
- The templates have been updated to not index (only display) the duplicated `data` types instead of just skipping them

#### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->](https://upload.wikimedia.org/wikipedia/commons/6/6e/Passer_domesticus_male_%2815%29.jpg)

SamWilsn commented on src/ethereum_spec_tools/autoapi_patch.py:
```diff
@@ -0,0 +1,61 @@
+Both instances are treated as `test.x` instead of `test.A.x` & `test.B.x`
+This could incorrectly lead to duplicates.
+
+This tool is uased as a patch.
```
```suggestion
This tool is used as a patch.
```

SamWilsn commented on src/ethereum_spec_tools/autoapi_patch.py:
```diff
@@ -0,0 +1,61 @@
+
+def get_assign_value_patch(node: Any) -> Any:
+    """
+    The get_assign_value function in autoapi.mappers.python.astroid_utils
+    does not handle Attribute assignment properly in that it only considers the
+    attribute name and omits out the information about the parent object
```
Would it be possible to upstream a patch?

gurukamath replied:
Yes. I do intend to raise a PR at some point. However, the patch itself introduces other kinds of duplicates which we know how to handle in the `specs` but would not know what to do with in a general context. I need to study those things deeper before upstreaming the patch.

Plus, I see that the `autoapi` repo is not updated that frequently. The last PR merge was in Feb with a  few still in queue. So it might take a few months.

I'll remove the `closes` tag from the PR so we can close the issue only after the patch is upstreamed

gurukamath commented:
@SamWilsn I have added the following general updates on top of this PR. Briefly

1. I see us building a few more little documentation related plug-ins/tools as we go forward and so, have separated out the documentation related `ethereum_spec_tools` out into its own package
2. Cleaned-up the API reference a bit and added it to the navigation sidebar. The docs for the different packages are now much easier to navigate
3. Re-structured the toctree so that the packages show up in the order of the ethereum hard forks instead of default alphabetical order.

SamWilsn commented on doc/_templates/autoapi/python/data.rst:
```diff
@@ -6,6 +6,9 @@
 
 {% if module is none %}
 .. py:{{ obj.type }}:: {{ obj.name }}
+   {%+ if obj.docstring == "autoapi_noindex" -%}
```
Can we use a contains check instead? I'd like to add documentation to constants eventually.

gurukamath replied:
Updated

SamWilsn commented on doc/_templates/autoapi/python/module.rst:
```diff
@@ -6,9 +6,6 @@
 :orphan:
 
 {% endif %}
-:mod:`{{ obj.name }}`
-======={{ "=" * obj.name|length }}
-
```
I think this removes the item from the TOC, right? Is that intentional?

gurukamath replied:
This does not remove the module from TOC, rather just the title. It will also display more meaningful titles which we have in the docstrings instead of just the module name

SamWilsn commented on setup.cfg:
```diff
@@ -99,7 +100,7 @@ ethereum =
     ethereum-spec-lint = ethereum_spec_tools.lint:main
     ethereum-spec-sync = ethereum_spec_tools.sync:main
-    ethereum-spec-diff = ethereum_spec_tools.diff:main
+    ethereum-spec-diff = ethereum_spec_tools.doc.diff:main
```
Hm... so far I've tried to keep a one-to-one correspondence between a single file/folder in `src/ethereum_spec_tools` and a single installed binary, to try to keep things understandable. I think I'd like to maintain that, at least until there are too many interdependencies.

gurukamath replied:
Ok. Updated

SamWilsn commented on doc/_templates/autoapi/python/data.rst:
```diff
@@ -34,11 +37,18 @@
     {% endif %}
 
 
+   {%+ if obj.docstring != "autoapi_noindex" -%}
```
I think all of these need to be updated to use `in` right?

gurukamath replied:
I retained this as `!=` so that if the doctoring is purely `autoapi_noindex`, nothing is rendered. 

However, as you mentioned earlier, going forward, we might have some useful information in these docstrings, in addition to the `autoapi_noindex` tag. 

If we change this to `in`, even the useful information will be missed.
